app/notifications/rest.py

```diff
@@ -233,23 +235,29 @@ def send_notification(notification_type):
 
     _service_allowed_to_send_to(notification, service)
 
-    saved_notification = None
-    if not _simulated_recipient(notification['to'], notification_type):
-        saved_notification = persist_notification(template_id=template.id,
-                                                  template_version=template.version,
-                                                  recipient=notification['to'],
-                                                  service=service,
-                                                  personalisation=notification.get('personalisation', None),
-                                                  notification_type=notification_type,
-                                                  api_key_id=api_user.id,
-                                                  key_type=api_user.key_type)
-        send_notification_to_queue(saved_notification, service.research_mode)
-
-    notification_id = create_uuid() if saved_notification is None else saved_notification.id
+    # Do not persist or send notification to the queue if it is a simulated recipient
+    simulated = simulated_recipient(notification['to'], notification_type)
+    saved_notification = persist_notification(template_id=template.id,
```

Comment:
Not sure I'm a fan of this. I like the abstraction but if I was reading this on first glance I'd automatically assume that the notification was persisted. 

I would have to zoom into `persist_notification` to know that it isn't.